src/protobuf.zig

```diff
@@ -963,6 +987,347 @@ pub fn pb_decode(comptime T: type, input: []const u8, allocator: Allocator) !T {
+                            union_field.type,
+                            allocator,
+                            source,
+                            options,
+                        ),
+                    );
+                    break;
+                }
+            } else return error.UnknownField;
+            if (.object_end != try source.next()) {
+                return error.UnexpectedToken;
+            }
+
+            break :oneof union_value;
+        },
+        .Varint, .FixedInt, .SubMessage, .String => other: {
+            // .SubMessage's (generated structs) and .String's
+            //   (ManagedString's) have its own jsonParse implementation
+            // Numeric types will be handled using default std.json parser
+            break :other try json.innerParse(
+                fieldInfo.type,
+                allocator,
+                source,
+                options,
+            );
+        },
+        // TODO: ATM there's no support for Timestamp, Duration
+        //   and some other protobuf types (see progress at
+        //   https://github.com/Arwalk/zig-protobuf/pull/49)
+        //   so it's better to see "switch must handle all possibilities"
+        //   compiler error here and then add JSON (de)serialization support
+        //   for them than hope that default std.json (de)serializer
+        //   will make all right by its own
+    };
+}
+
+pub fn pb_json_decode(
+    comptime T: type,
+    input: []const u8,
+    options: json.ParseOptions,
+    allocator: Allocator,
+) !std.json.Parsed(T) {
+    const parsed = try json.parseFromSlice(T, allocator, input, options);
+    return parsed;
+}
+
+pub fn pb_json_encode(
+    data: anytype,
+    options: json.StringifyOptions,
+    allocator: Allocator,
+) ![]u8 {
+    return try json.stringifyAlloc(allocator, data, options);
+}
+
+fn to_camel_case(not_camel_cased_string: []const u8, allocator: Allocator) ![]const u8 {
+    var underscore_count: usize = 0;
+    for (not_camel_cased_string) |byte| {
+        if (byte == '_') underscore_count += 1;
+    }
+    if (underscore_count == 0 or underscore_count == not_camel_cased_string.len) {
+        // You better not touch fields like "_____________" (or should we?)
+        return not_camel_cased_string;
+    }
+
+    var capitalize_next_letter = false;
+    var camel_cased_string = try allocator.alloc(u8, not_camel_cased_string.len - underscore_count);
+
+    var i: usize = 0;
+    for (not_camel_cased_string) |char| {
+        if (char == '_') {
+            capitalize_next_letter = i > 0;
+        } else if (capitalize_next_letter) {
+            camel_cased_string[i] = std.ascii.toUpper(char);
+            capitalize_next_letter = false;
+            i += 1;
+        } else {
+            camel_cased_string[i] = char;
+            i += 1;
+        }
+    }
+
+    if (std.ascii.isUpper(camel_cased_string[0])) {
+        camel_cased_string[0] = std.ascii.toLower(camel_cased_string[0]);
```

Comment:
This misses the case where the string starts with `_` then a capital letter; is that on purpose?

Reply:
I just took the implementation from [this python code](https://github.com/protocolbuffers/protobuf/blob/06a520cdb78be66f506b88a8adba2763f2301dc0/python/google/protobuf/descriptor.py#L1299) and also [this C++ code](https://github.com/protocolbuffers/protobuf/blob/06a520cdb78be66f506b88a8adba2763f2301dc0/hpb_generator/gen_messages.cc#L126). Since it's the official google libraries others supported languages probably have the same conversion code.

Reply:
In case you're talking about it'll skip all leading underscores and then will make first letter the lowercase one